src/ord.rs

```diff
@@ -35,47 +35,24 @@ impl fmt::Display for EqOps {
 ///
 /// This is created by the `predicate::{eq, ne}` functions.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct EqPredicate<T>
-where
-    T: fmt::Debug + PartialEq,
-{
+pub struct EqPredicate<T> {
     constant: T,
     op: EqOps,
 }
 
-impl<T> Predicate<T> for EqPredicate<T>
-where
-    T: fmt::Debug + PartialEq,
-{
-    fn eval(&self, variable: &T) -> bool {
-        match self.op {
-            EqOps::Equal => variable.eq(&self.constant),
-            EqOps::NotEqual => variable.ne(&self.constant),
-        }
-    }
-
-    fn find_case<'a>(&'a self, expected: bool, variable: &T) -> Option<reflection::Case<'a>> {
-        utils::default_find_case(self, expected, variable).map(|case| {
-            case.add_product(reflection::Product::new(
-                "var",
-                utils::DebugAdapter::new(variable).to_string(),
-            ))
-        })
-    }
-}
-
-impl<'a, T> Predicate<T> for EqPredicate<&'a T>
+impl<T, P> Predicate<T> for EqPredicate<P>
```

Comment:
imo using different generic parameters between blocks is confusing, please make them consistent (`EqPredicate<T>` vs `EqPredicate<P>`).  Ideally anything deviating from the existing would be done in its own commit before this one or in a whole PR before this one,

Reply:
Ok, I can invert the use of predicate parameters here and below.